File: app/src/main/java/org/worshipsongs/parser/ISongParser.java
package org.worshipsongs.parser;

import android.content.Context;

import org.worshipsongs.domain.Verse;

import java.util.List;

/**
 * Author : Madasamy
 * Version : 3.x
 */

public interface ISongParser
{

    List<String> parseContents(Context context, String lyrics, String verseOrder);

    List<Verse> parseVerse(Context context, String lyrics);

    List<String> getVerseOrders(String verseOrder);

    String parseMediaUrlKey(String comments);

    String parseChord(String comments);

    String parseTamilTitle(String comments);

}

File: app/src/main/java/org/worshipsongs/parser/ISongParser.kt
package org.worshipsongs.parser

import android.content.Context

import org.worshipsongs.domain.Verse

/**
 * Author : Madasamy
 * Version : 3.x
 */

interface ISongParser
{

    fun parseContents(context: Context, lyrics: String, verseOrder: String): List<String>

    fun parseVerse(context: Context, lyrics: String): List<Verse>

    fun getVerseOrders(verseOrder: String): List<String>

    fun parseMediaUrlKey(comments: String?): String

    fun parseChord(comments: String?): String

    fun parseTamilTitle(comments: String?): String

}
